src/components/editor/ComponentPreview.tsx

```diff
@@ -43,9 +52,31 @@ const ComponentPreview: React.FC<{
   if (!component) {
     console.error(`ComponentPreview unavailable for component ${componentName}`)
   }
-
   const type = (component && component.type) || null
 
+  const [views, setViews] = useState<any>([])
+  const customComponents = useSelector(getCustomComponentNames)
+
+  useEffect(() => {
+    async function loadViews() {
+      const componentPromises = await customComponents.map(
+        async (customComponent: string) => {
+          const View = await importView(customComponent)
+          return <View key={customComponent} component={component} />
+        },
+      )
+      Promise.all(componentPromises).then(setViews)
+    }
+    loadViews()
+  }, [customComponents])
```

Comment:
Just names or component jsons?

Reply:
just the names

Reply:
shouldn't it update with custom components json as well

Reply:
It is not needed, this function is only responsible for importing the custom components being worked on, so it should only change when the component names change, so far as those components are already imported, any change made to any of them will be reflected